src/main/java/io/jenkins/plugins/opentelemetry/job/log/BuildInfo.java

```diff
@@ -5,38 +5,70 @@
 
 package io.jenkins.plugins.opentelemetry.job.log;
 
+import com.google.common.base.Objects;
 import io.jenkins.plugins.opentelemetry.semconv.JenkinsOtelSemanticAttributes;
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.context.Context;
+import org.yaml.snakeyaml.nodes.CollectionNode;
 
+import javax.annotation.Nonnull;
+import java.util.Collections;
 import java.util.Map;
 
-final class BuildInfo {
+public final class BuildInfo {
```

Comment:
In Otel code, they have the guiding principle to keep as much things as possible package protected and final. I like this. It's why I used package protected. Let's see what we want to do.